test/index.js

```diff
@@ -1961,4 +1961,58 @@ describe('h2o2', () => {
 
         expect(res.payload).to.equal('ok');
     });
+
+    it('propagates cancellation to upstream', { parallel: false }, async () => {
+
+        let notifyRequestReceived;
+        const upstreamRequestReceived = new Promise((resolve) => {
+
+            notifyRequestReceived = resolve;
+        });
+
+        let notifyRequestAborted;
+        const upstreamRequestAborted = new Promise((resolve) => {
+
+            notifyRequestAborted = resolve;
+        });
+
+        const upstream = Hapi.server();
+        upstream.route({
+            method: 'GET',
+            path: '/cancellation',
+            handler: function (request, h) {
+
+                notifyRequestReceived();
+
+                request.raw.req.once('aborted', () => {
+
+                    notifyRequestAborted(true);
+                });
+
+                return new Promise(() => {
+
+                    return h.response('never resolves');
+                });
+            }
+        });
+        await upstream.start();
+
+        const server = Hapi.server();
+        await server.register(H2o2);
+        server.route({ method: 'GET', path: '/cancellation', handler: { proxy: { host: 'localhost', port: upstream.info.port } } });
+        await server.start();
+
+        const promise = Wreck.request('GET', `http://${server.info.host}:${server.info.port}/cancellation`);
+        const req = promise.req;
+        promise.catch(() => undefined);
+
+        await upstreamRequestReceived;
```

Comment:
We need to wait until the proxy request is effectively performed before canceling the request, else the upstream handler will just not be invoked.